tests/pfcwd/test_pfcwd_function.py

```diff
@@ -551,3 +623,62 @@ def test_pfcwd_actions(self, request, fake_storm, setup_pfc_test, fanout_graph_f
                      logger.info("--- Stop PFC WD ---")
                      self.dut.command("pfcwd stop")
+
+    def test_pfcwd_mmu_change(self, request, fake_storm, setup_pfc_test, fanout_graph_facts, ptfhost, duthosts, rand_one_dut_hostname, fanouthosts):
+        """
+        Tests if mmu changes impact Pfcwd functionality
+
+        Args:
+            request(object) : pytest request object
+            fake_storm(fixture) : Module scoped fixture for enable/disable fake storm
+            setup_pfc_test(fixture) : Module scoped autouse fixture for PFCwd
+            fanout_graph_facts(fixture) : fanout graph info
+            ptfhost(AnsibleHost) : ptf host instance
+            duthost(AnsibleHost) : DUT instance
+            rand_one_dut_hostname(string) : randomly pick a dut in multi DUT setup
+            fanouthosts(AnsibleHost): fanout instance
+        """
+        duthost = duthosts[rand_one_dut_hostname]
+        setup_info = setup_pfc_test
+        self.fanout_info = fanout_graph_facts
+        self.ptf = ptfhost
+        self.dut = duthost
+        self.fanout = fanouthosts
+        self.timers = setup_info['pfc_timers']
+        self.ports = setup_info['selected_test_ports']
+        key, value  = self.ports.items()[0]
+        self.ports = {key: value}
+        port = key
+        self.neighbors = setup_info['neighbors']
+        dut_facts = self.dut.facts
+        self.peer_dev_list = dict()
+        self.fake_storm = fake_storm
+        self.storm_hndle = None
+        logger.info("---- Testing on port {} ----".format(port))
+        self.setup_test_params(port, setup_info['vlan'], init=True, mmu_params=True)
+        self.stats = PfcPktCntrs(self.dut, "drop")
+
+        try:
+            for idx, mmu_action in enumerate(MMU_ACTIONS):
+                self.traffic_inst = SendVerifyTraffic(self.ptf, dut_facts['router_mac'], self.pfc_wd)
+                pfc_wd_restore_time_large = request.config.getoption("--restore-time")
+                # wait time before we check the logs for the 'restore' signature. 'pfc_wd_restore_time_large' is in ms.
+                self.timers['pfc_wd_wait_for_restore_time'] = int(pfc_wd_restore_time_large / 1000 * 2)
+                if idx:
+                    self.update_queue(port)
+                    self.storm_setup()
+                self.traffic_inst = SendVerifyTraffic(self.ptf, dut_facts['router_mac'], self.pfc_wd)
+                self.run_test(self.dut, port, "drop", mmu_action=mmu_action)
+                self.dut.command("pfcwd stop")
+
+        except Exception as e:
+            pytest.fail(str(e))
+
+        finally:
+            if self.storm_hndle:
+                logger.info("--- Stop pfc storm on port {}".format(port))
+                self.storm_hndle.stop_storm()
+            # restore alpha
+            PfcCmd.update_alpha(self.dut, self.pg_profile, self.alpha)
+            logger.info("--- Stop PFC WD ---")
+            self.dut.command("pfcwd stop")
```

Comment:
Why stop PFC watchdog at the end?

Reply:
For each iteration, we need to stop pfcwd. but when this module is done, there is a module level fixture which ensure that pfcwd is started with default settings